pages/pil/exemptions/routers/modules.js

```diff
@@ -1,42 +1,68 @@
 const { Router } = require('express');
+const { omit, castArray } = require('lodash');
+
 const form = require('../../../common/routers/form');
+const { buildModel } = require('../../../../lib/utils');
+
 const { modules: schema } = require('../schema');
 const { moduleCodes } = require('@asl/constants');
+const { species } = require('@asl/constants');
+const content = require('../content/modules');
+
+const modulesThatRequireSpecies = [
+  'PILA (theory)',
+  'PILA (skills)',
+  'K (theory)',
+  'K (skills)'
+];
 
 module.exports = () => {
   const app = Router();
 
   app.use('/', (req, res, next) => {
     const exemptions = req.profile.exemptions;
-
-    req.model = {
-      modules: []
-    };
+    req.model = buildModel(schema);
 
     req.model = exemptions.reduce((obj, value, key) => {
       const module = value.module;
       return {
         ...obj,
         modules: [ ...obj.modules, module ],
-        [`module-${module}-reason`]: value.exemptionDescription
+        [`module-${module}-reason`]: value.exemptionDescription,
+        [`module-${module}-species`]: value.species
       };
     }, req.model);
 
     next();
   });
 
   app.use('/', form({
-    schema: {
-      ...schema,
-      ...schema.modules.options.reduce((obj, val) => {
-        return {
-          ...obj,
-          [`module-${val.value}-reason`]: val.reveal.reason
-        };
-      }, {})
+    configure: (req, res, next) => {
+      req.form.schema = {
+        ...schema,
+        ...schema.modules.options.reduce((obj, val) => {
+
+          let type = {
+            ...obj,
+            [`module-${val.value}-reason`]: val.reveal.reason
+          };
+
+          if (modulesThatRequireSpecies.includes(val.value)) {
+            type[`module-${val.value}-species`] = {
+              inputType: 'select',
+              options: species,
+              label: content.fields.species.label
+            };
+          }
+
+          return type;
+        }, {})
+      };
+      next();
     },
     locals: (req, res, next) => {
-      res.locals.static.schema = schema;
+      res.locals.static.schema = omit(req.form.schema, moduleCodes.map(c => `module-${c}-reason`).concat(moduleCodes.map(c => `module-${c}-species`)));
```

Comment:
I have no idea what this line of code is doing apart from "too much". Can we split it out a little to make it clearer?